sdks/java/extensions/python/src/main/resources/org/apache/beam/sdk/extensions/python/bootstrap_beam_venv.py

```diff
@@ -64,13 +63,13 @@ def main():
         info = json.load(
             urllib.request.urlopen("https://pypi.org/pypi/apache_beam/json"))
 
-        def maybe_strict_version(s):
-            try:
-                return parse_version(s)
-            except:
-                return parse_version('0.0')
+        def released_version(s):
```

Comment:
I have a simpler fix

Reply:
sent https://github.com/apache/beam/pull/31487, PTAL and merge if this looks good or feel free to take over that PR.